src/Access/SettingsConstraints.cpp

```diff
@@ -36,9 +36,43 @@ void SettingsConstraints::clear()
     constraints.clear();
 }
 
+namespace
+{
+
+std::string_view resolveName(std::string_view name)
+{
+    auto write_warning_for_changed = [](const auto resolved_name, const auto original_name)
+    {
+        LOG_WARNING(
+            &Poco::Logger::get("SettingsConstaints"),
+            "Setting '{}' was renamed to '{}', please use '{}' when defining constraints",
+            original_name,
+            resolved_name,
+            resolved_name);
+    };
+
+    if (auto resolved_name = SettingsTraits::resolveName(name); resolved_name != name)
+    {
+        write_warning_for_changed(resolved_name, name);
+        return resolved_name;
+    }
+
+    if (auto resolved_name = MergeTreeSettingsTraits::resolveName(name); resolved_name != name)
+    {
+        write_warning_for_changed(resolved_name, name);
+        return resolved_name;
+    }
```

Comment:
This is a little bit wrong. It does not take `MERGE_TREE_SETTINGS_PREFIX` into account. I think it may be better to integrate this logic into `resolveSetting.h`.